oneview_redfish_toolkit/api/manager.py

```diff
@@ -27,31 +32,51 @@ class Manager(RedfishJsonValidator):
 
     SCHEMA_NAME = 'Manager'
 
-    def __init__(self, oneview_resource, firmware_version):
+    def __init__(self, oneview_appliance_info, oneview_appliance_state,
+                 oneview_appliance_health_status):
         """Manager constructor
 
             Populates self.redfish with some hardcoded Manager
             values and with the response from OneView.
 
             Args:
-                oneview_resource: A dict for ServerHardware or Enclosure
+                oneview_appliance_info: An Oneview's appliance info dict
+                oneview_appliance_state: An Oneview's appliance status dict
+                oneview_appliance_health_status: An Oneview's appliance
+                health state dict
         """
 
         super().__init__(self.SCHEMA_NAME)
 
         self.redfish["@odata.type"] = self.get_odata_type()
-        self.redfish["Id"] = oneview_resource['uuid']
-        self.redfish["Description"] = None
-        self.redfish["FirmwareVersion"] = firmware_version
-
-        # TODO(victorhugorodrigues): update state and health properties
-        # self.redfish["Status"] = collections.OrderedDict()
-        # state, _ = status_mapping.\
-        #     get_redfish_server_hardware_status_struct(oneview_resource)
-        # self.redfish["Status"]["State"] = state
-        # self.redfish["Status"]["Health"] = health
-
+        self.redfish["Id"] = oneview_appliance_info['uuid']
+        self.redfish["Description"] = oneview_appliance_info["family"]
+        self.redfish["ManagerType"] = "Service"
+        self.redfish["FirmwareVersion"] = \
+            oneview_appliance_info["softwareVersion"]
+        self.redfish["Status"] = collections.OrderedDict()
+        state = status_mapping.APPLIANCE_STATE_TO_REDFISH_STATE_MAPPING.\
+            get(oneview_appliance_state["state"])
+        health = self._get_highest_health_state(
+            oneview_appliance_health_status["members"])
+        self.redfish["Status"]["State"] = state
+        self.redfish["Status"]["Health"] = health
         self.redfish["@odata.context"] = \
             "/redfish/v1/$metadata#Manager.Manager"
         self.redfish["@odata.id"] = \
-            "/redfish/v1/Managers/" + oneview_resource['uuid']
+            "/redfish/v1/Managers/" + oneview_appliance_info['uuid']
+
+    @staticmethod
+    def _get_highest_health_state(health_state_members):
+        health_states = dict()
+
+        for member in health_state_members:
+            redfish_health_state = status_mapping.MANAGER_HEALTH_STATE_MAPPING.\
+                get(member["severity"])
+            health_states[redfish_health_state] = \
```

Comment:
Would this work and simplify if we replace by:

highest_status = \
            max(highest_status, status_mapping.CRITICALITY_STATUS_MAPPING[redfish_health_state])

Seems we can rename all mapping structures that are already in status_mapping, e.g. "CRITICALITY_STATUS"

Reply:
@adriane-cardozo I understand your suggestion for highest_status but the health_state_members list (with 3 members) is composed by the values that we need first, map to the redfish state, as also check between these three values which of them has the most critical state. Because of that, we map each member (severity) to redfish state and then I assigned them to health_states list so that in this way I can check what is the highest status.

Regarding his second remark, I will rename all mapping structures removing "MAPPING" word for each map that has it.